src/plugins/content_management/demo/todo/todos_client.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+import type { CrudClient } from '../../public/crud_client';
+import type { CreateIn, DeleteIn, GetIn, SearchIn, SearchOut, UpdateIn } from '../../common';
+
+export interface Todo {
+  id: string;
+  title: string;
+  completed: boolean;
+}
+
+export class TodosClient implements CrudClient {
+  private todos: Todo[] = [
+    { id: uuidv4(), title: 'Learn Elasticsearch', completed: true },
+    { id: uuidv4(), title: 'Learn Kibana', completed: false },
+  ];
+
+  async create<I extends CreateIn = CreateIn, O = unknown>(input: I): Promise<O> {
```

Comment:
Current types approach didn't work quite well. Ideally I wanted to define these types to be `Todo` specific, like: 

`async create(input: TodoCreateIn): Promise<TodoCreateOut> {`

but using the `export class TodosClient implements CrudClient` with the current `CrudClient` I couldn't do it: 


```
TS2416: Property 'create' in type 'TodosClient' is not assignable to the same property in base type 'CrudClient'.   Type '(input: { data: Todo; }) => Promise<Todo>' is not assignable to type '<I extends CreateIn<string, Record<string, unknown>, any> = CreateIn<string, Record<string, unknown>, any>, O = unknown>(input: I) => Promise<O>'.     Types of parameters 'input' and 'input' are incompatible.       Type 'I' is not assignable to type '{ data: Todo; }'.         Type 'CreateIn<string, Record<string, unknown>, any>' is not assignable to type '{ data: Todo; }'.           Types of property 'data' are incompatible.             Type 'Record<string, unknown>' is missing the following properties from type 'Todo': id, title, completed
```

Reply:
This is resolved by removing generics on CrudClient